x-pack/plugins/fleet/server/services/agents/reassign.ts

```diff
@@ -62,7 +88,15 @@ export async function reassignAgents(
             showInactive: false,
           })
         ).agents;
-  const agentsToUpdate = agents.filter((agent) => agent.policy_id !== newAgentPolicyId);
+  // And which are allowed to unenroll
```

Comment:
Curious to have your thought on how this will scale, I am wondering if should do this in multiple phase:
* collect the policy ids we need
* create a map of <policy_id,is_managed>
* than filtering the agents

if we have 1000 agents selected here otherwise it's 1000 concurrent request to ES

Reply:
Thanks for 👀 . 🎯 about the many requests. I think I'll change it to do a `bulkGet` for the agent policies and use that for the map like you said.

I've put off changing the `*isAllowed` function to accept either a string or object, but this is a good reason to change that.

Reply:
@nchaulet I'd like to do this in a follow up PR if that's ok. I'd like to get the existing features merged in so others can use them. I'll come back to this as part of the changes to bulk operations (having them error instead of filter out invalid values)

Reply:
Sure

Reply:
@nchaulet sorry, I missed this yesterday but I did add this to https://github.com/elastic/kibana/issues/90437